### PR TITLE
feat: add `last_activity_at` to feedback datasets

### LINKED ISSUE #3981: [FEATURE] Add `last_activity_at` to Feedback Datasets
Right now feedback datasets are exposing internal database `updated_at` attribute with the timestamp of the moment the dataset was updated.

This is not really useful for users so we want to add a new `last_activity_at` attribute to feedback datasets with the timestamp of the last updated response.

`last_activity_at` will be updated with the following cases:
* The dataset has been created.
* The dataset has been updated.
* A response for a record belonging to the dataset has been created.
* A response for a record belonging to the dataset has been updated.
* A response for a record belonging to the dataset has been deleted.

This feature will require changes on the frontend to use `last_activity_at` instead of `updated_at` for feedback datasets. 

/cc @damianpumar

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR adds a new `DateTime` column to `datasets` table with name `last_activity_at`.

This new column will be updated for the following cases:
* When a dataset is created or updated.
* When a response for a record belonging to the dataset has been created, updated or deleted.

The idea is that we can use `last_activity_at` to record the last time that an activity associated with a dataset happened and use it instead of `updated_at` to show recent activity on the UI.

Closes #3981

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding new tests and running them locally.
- [x] Running tests using PostgreSQL locally.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by jfcalvo** [comment]
@damianpumar please feel free to add here the frontend changes, using this new dataset attribute instead of `updated_at` for feedback datasets.

Once it's done please set the PR as ready for review.

**Comment by damianpumar** [comment]
> @damianpumar please feel free to add here the frontend changes, using this new dataset attribute instead of `updated_at` for feedback datasets.
> 
> Once it's done please set the PR as ready for review.

From my side is done!

File: src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# revision identifiers, used by Alembic.
+revision = "26721800a9f3"
+down_revision = "1e629a913727"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("datasets", sa.Column("last_activity_at", sa.DateTime(), nullable=False))
```
Comment (gabrielmbmb) [inline src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py]: for datasets that were created before this migration is applied, we have to give a default value if `nullable=False`, right?

File: src/argilla/server/models/models.py [hunk 2]
```diff
@@ -170,13 +173,22 @@ def __repr__(self):
...
 class Dataset(DatabaseModel):
     __tablename__ = "datasets"
 
     name: Mapped[str] = mapped_column(index=True)
     guidelines: Mapped[Optional[str]] = mapped_column(Text)
     status: Mapped[DatasetStatus] = mapped_column(DatasetStatusEnum, default=DatasetStatus.draft, index=True)
     workspace_id: Mapped[UUID] = mapped_column(ForeignKey("workspaces.id", ondelete="CASCADE"), index=True)
+    inserted_at: Mapped[datetime] = mapped_column(default=datetime.utcnow)
+    updated_at: Mapped[datetime] = mapped_column(default=inserted_at_current_value, onupdate=datetime.utcnow)
+    last_activity_at: Mapped[datetime] = mapped_column(
```
Comment (frascuchon) [inline src/argilla/server/models/models.py]: What would be the value for existing datasets? Do we need to do an extra migration step?

Comment (jfcalvo) [reply src/argilla/server/models/models.py]: Yes, I totally miss that. We can copy the value from `updated_at` or set it as optional.

File: src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py]: Yes, I miss that. We need to set a default value, migrate old datasets or set it as optional.

File: src/argilla/server/models/models.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla/server/models/models.py]: Done. I updated the migration to use `updated_at` value to populate the new column.

File: src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/alembic/versions/26721800a9f3_add_last_activity_at_to_datasets_table.py]: Done. I updated the migration to use `updated_at` value to populate the new column.

**Review by gabrielmbmb — APPROVED** [review]

**Review by frascuchon — APPROVED** [review]

**Pull request merged**